### ISSUE TITLE
Create credentials CRD

### ISSUE THREAD

**Issue by JorritSalverda** [estafette/estafette-ci-operator#1]
Support a credential CRD with the following api version and kind

```
apiVersion: ci.estafette.io/v1
kind: credentials
```
And the following fields in the spec:

```
name: <string, required and needs to be unique>
type: <string, required>
whitelistedTrustedImages: <string, optional>
whitelistedPipelines: <string, optional>

additionalProperties: <map[string]interface{}, optional>
```

See https://github.com/estafette/estafette-ci-contracts/blob/master/config.go#L43-L51

For the additional properties it would be nice if they can be inlined like we already do in the configmap, but not sure if that's possible since kubebuilder tries to create protos for the CRDs.

**Comment by vitorarins** [comment]
Yeah, kubebuilder doesn't even allow map[string]interface{} to be used.

I've found this: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/de0b9e0704/examples/dashboard-operator/api/v1alpha1/dashboard_types.go#L28
but in our case we will need to define the fields that could go into the additional properties.

**Comment by vitorarins** [comment]
Maybe we can have one CRD per credential type, ex.: registrycredential, slackcredential, etc.

**Comment by JorritSalverda** [comment]
The reason they're not strongly typed is that anyone can come up with new types belonging to some extension container without Estafette having any knowledge of the type whatsoever. Making a CRD per type renders that impossible. I'm up for ways to make it easier to validate the properties, while keeping the freedom to add new types without touching the core of Estafette.

**Comment by vitorarins** [comment]
got it, makes sense :+1:

**Comment by vitorarins** [comment]
Fixed by #7
